agent/hcp/client/metrics_client.go

```diff
@@ -0,0 +1,165 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"golang.org/x/oauth2"
+	"google.golang.org/protobuf/proto"
+
+	colmetricpb "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	metricpb "go.opentelemetry.io/proto/otlp/metrics/v1"
+
+	"github.com/hashicorp/consul/version"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-retryablehttp"
+	hcpcfg "github.com/hashicorp/hcp-sdk-go/config"
+)
+
+const (
+	// HTTP Client config
+	defaultStreamTimeout = 15 * time.Second
+
+	// Retry config
+	defaultRetryWaitMin = 1 * time.Second
+	defaultRetryWaitMax = 15 * time.Second
+	defaultRetryMax     = 4
+)
+
+// MetricsClient exports Consul metrics in OTLP format to the HCP Telemetry Gateway.
+type MetricsClient interface {
+	ExportMetrics(ctx context.Context, protoMetrics *metricpb.ResourceMetrics, endpoint string) error
+}
+
+// hcpConfig represents HCP config for TLS abstracted in an interface for easy testing.
+type hcpConfig interface {
+	oauth2.TokenSource
+	APITLSConfig() *tls.Config
+}
+
+// cloudConfig represents cloud config for TLS abstracted in an interface for easy testing.
+type cloudConfig interface {
+	HCPConfig(opts ...hcpcfg.HCPConfigOption) (hcpConfig, error)
+}
+
+// otlpClient is an implementation of MetricsClient with a retryable http client for retries and to honor throttle.
+// It also holds default HTTP headers to add to export requests.
+type otlpClient struct {
+	client  *retryablehttp.Client
+	headers map[string]string
+}
+
+// TelemetryClientCfg is used to configure the MetricsClient.
+type TelemetryClientCfg struct {
+	CloudCfg cloudConfig
+	Logger   hclog.Logger
+}
+
+// NewMetricsClient returns a configured MetricsClient.
+// The current implementation uses otlpClient to provide retry functionality.
+func NewMetricsClient(cfg *TelemetryClientCfg) (MetricsClient, error) {
+	if cfg.CloudCfg == nil || cfg.Logger == nil {
+		return nil, fmt.Errorf("failed to init telemetry client: provide valid TelemetryClientCfg")
+	}
+
+	c, err := newHTTPClient(cfg.CloudCfg, cfg.Logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to init telemetry client: %v", err)
+	}
+
+	headers := map[string]string{
+		"X-HCP-Source-Channel": fmt.Sprintf("consul %s hcp-go-sdk/%s", version.GetHumanVersion(), version.Version),
```

Comment:
These are some default headers that are needed, let me know if I'm missing any others.  

 I looked at this [client](https://github.com/hashicorp/hcp-sdk-go/blob/main/httpclient/httpclient.go) for inspiration (used in the other hcp client)

Reply:
@nickethier mentioned missing `X-Resource-ID`? Do you have an example of how this is set?

Reply:
The go sdk isn't really used, so consider removing that.

Reply:
As mentioned on the call earlier, if the telemetry gateway isn't reading and acting on any of these headers then maybe we can just remove them?

Reply:
Do we want to completely remove it or just the `hcp-go-sdk`? I thought that it was a standard and if not already, eventually useful for internal purposes (that header acts like a user agent from what I remember in our discussion). @nickethier any thoughts on how the Gateway uses these?

Happy to remove if useless